xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -1631,6 +1684,116 @@ bool CGUIDialogVideoInfo::SetMovieSet(const CFileItem *movieItem, const CFileIte
   return true;
 }
 
+bool CGUIDialogVideoInfo::SetMovieVersion(const CFileItemPtr &item)
+{
+  // Set this movie as an alternate version of another movie, user will
+  // be asked to provide the target movie and the alternate version type
+  // (EXTENDED, DIRECTORS CUT, etc.), this movie entry will be deleted
+  // from DB at last
+  //
+  // N.B. Choosing the movie itself as target movie is also allowed, which
+  //      will just change the alternate version type if that get changed
+
+  if (item == nullptr || !item->HasVideoInfoTag())
+    return false;
+
+  CVideoDatabase videodb;
+  if (!videodb.Open())
+    return false;
+
+  CFileItemList list;
+  int dbId = item->GetVideoInfoTag()->m_iDbId;
+
+  videodb.GetMovieVersion(dbId, list);
+  if (list.Size() > 1)
+  {
+      CGUIDialogOK* dialog = CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogOK>(WINDOW_DIALOG_OK);
+      if (!dialog)
+        return false;
+
+      dialog->SetHeading(CVariant{39305});
+      dialog->SetLine(1, CVariant{39306});
+      dialog->Open();
+
+      return false;
+  }
+
+  list.Clear();
+
+  int targetDbId;
+
+  // choose the target movie
+  videodb.GetMoviesNav("videodb://movies/titles", list);
+  if (list.Size() < 1)
+    return false;
+
+  int itemId = 0;
+
+  for (int i = 0; i < list.Size(); i++)
+  {
+    if (list[i]->GetVideoInfoTag()->m_iDbId == dbId)
+    {
+      itemId = i;
+      break;
+    }
+  }
+
+  CGUIDialogSelect* dialog = CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(WINDOW_DIALOG_SELECT);
+  if (!dialog)
+    return false;
+
+  int selected = -1;
+
+  list.Sort(SortByLabel, SortOrderAscending, CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_FILELISTS_IGNORETHEWHENSORTING) ? SortAttributeIgnoreArticle : SortAttributeNone);
+  dialog->Reset();
+  dialog->SetItems(list);
+  dialog->SetHeading(CVariant{39302});
+  dialog->SetSelected(itemId);
+  dialog->Open();
+  selected = dialog->GetSelectedItem();
```

Comment:
Why not declare and init the var in one step here? => `int selected = dialog...`